Fix envelope.ts to use new error system properties

@claude Please update envelope.ts to use the new enhanced error system properties

**Problem:**
Lines 207 and 209 in `src/envelope.ts` are trying to access `error.details` and `error.notionError` properties which existed in the legacy error system but don't exist in the new NotionCLIError class.

**Compilation errors:**
```
src/envelope.ts(207,29): error TS2339: Property 'details' does not exist on type 'NotionCLIError'.
src/envelope.ts(209,28): error TS2339: Property 'notionError' does not exist on type 'NotionCLIError'.
```

**New error system properties:**
The new NotionCLIError has:
- `code: NotionCLIErrorCode` (not ErrorCode)
- `userMessage: string` (the main message)
- `suggestions: ErrorSuggestion[]`
- `context: ErrorContext` (this contains metadata, attemptedId, originalError, etc.)
- `timestamp: string`

**Tasks:**
1. Read `src/errors/enhanced-errors.ts` to understand the new structure
2. Update lines 207-210 in `src/envelope.ts` to use the new properties
3. Replace `error.details` with `error.context` (or appropriate property)
4. Replace `error.notionError` with `error.context.originalError`
5. Ensure the envelope still logs the necessary debug information

**Files to modify:**
- `src/envelope.ts` (update error property access)

**Reference:**
- New error system definition: `src/errors/enhanced-errors.ts` lines 104-195

--- a/src/envelope.ts
+++ b/src/envelope.ts
@@ -204,9 +204,9 @@ export class EnvelopeFormatter {
       errorDetails = {
         code: error.code,
         message: error.message,
-        details: { ...error.details, ...additionalContext },
-        suggestions: generateSuggestions(error.code),
-        notionError: error.notionError,
+        details: { ...error.context, ...additionalContext },
+        suggestions: error.suggestions.map(s => s.description),
+        notionError: error.context.originalError,
       }
     }
     // Handle standard Error

